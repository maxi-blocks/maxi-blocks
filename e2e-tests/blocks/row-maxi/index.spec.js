/**
 * WordPress dependencies
 */
import { createNewPost } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	getBlockStyle,
	addCustomCSS,
	getEditedPostContent,
	insertMaxiBlock,
	updateAllBlockUniqueIds,
} from '../../utils';

describe('Row Maxi', () => {
	it('Row Maxi does not break', async () => {
		await createNewPost();
		await insertMaxiBlock(page, 'Container Maxi');
		await updateAllBlockUniqueIds(page);

		await page.waitForSelector('.maxi-row-block__template button');
		await page.waitForTimeout(100);
		await page.$$eval('.maxi-row-block__template button', button =>
			button[1].click()
		);
		await page.waitForSelector('.maxi-column-block');

		await updateAllBlockUniqueIds(page);

		expect(await getEditedPostContent(page)).toMatchSnapshot();
		expect(await getBlockStyle(page)).toMatchSnapshot();
	});
	it('Row Maxi Custom CSS', async () => {
		await expect(await addCustomCSS(page)).toMatchSnapshot();
	}, 500000);
});
