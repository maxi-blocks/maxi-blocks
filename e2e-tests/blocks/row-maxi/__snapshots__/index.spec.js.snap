// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Maxi Row Maxi Custom CSS 1`] = `"<style>body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 1170px; flex-wrap: wrap; row-gap: 20px; column-gap: 2.5%; background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xl\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 1170px; flex-wrap: wrap; row-gap: 20px; column-gap: 2.5%; background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.row-maxi-1::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.row-maxi-1::before{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xl\"] .maxi-block.maxi-block--backend.row-maxi-1::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1::before{ background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.row-maxi-1::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.row-maxi-1::after{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xl\"] .maxi-block.maxi-block--backend.row-maxi-1::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1::after{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 90%;}</style>"`;

exports[`Row Maxi Row Maxi does not break 1`] = `
"<!-- wp:maxi-blocks/container-maxi {\"maxi-version-current\":\"1.3.1\",\"maxi-version-origin\":\"1.3.1\",\"blockStyle\":\"light\",\"isFirstOnHierarchy\":true,\"uniqueID\":\"container-maxi-1\",\"customLabel\":\"Container_1\",\"max-width-unit-xxl\":\"px\",\"max-width-unit-xl\":\"px\",\"max-width-unit-l\":\"%\",\"max-width-xxl\":\"1690\",\"max-width-xl\":\"1170\",\"max-width-l\":\"90\",\"width-unit-l\":\"px\",\"width-l\":\"1170\",\"width-m\":\"1000\",\"width-s\":\"700\",\"width-xs\":\"460\"} -->
<section class=\"wp-block-maxi-blocks-container-maxi maxi-block maxi-container-block maxi-light container-maxi-1\" id=\"container-maxi-1\"><div class=\"maxi-background-displayer\"><div class=\"maxi-background-displayer__layer maxi-background-displayer__1\"></div></div><!-- wp:maxi-blocks/row-maxi {\"maxi-version-current\":\"1.3.1\",\"maxi-version-origin\":\"1.3.1\",\"blockStyle\":\"light\",\"isFirstOnHierarchy\":false,\"uniqueID\":\"row-maxi-1\",\"customLabel\":\"Row_1\",\"row-pattern-general\":\"1-1\",\"row-pattern-m\":\"2-2\",\"max-width-unit-xxl\":\"px\",\"max-width-unit-xl\":\"px\",\"max-width-unit-l\":\"%\",\"max-width-xxl\":\"1690\",\"max-width-xl\":\"1170\",\"max-width-l\":\"90\",\"flex-wrap-general\":\"wrap\"} -->
<div class=\"wp-block-maxi-blocks-row-maxi maxi-block maxi-row-block maxi-light row-maxi-1\" id=\"row-maxi-1\"><!-- wp:maxi-blocks/column-maxi {\"maxi-version-current\":\"1.3.1\",\"maxi-version-origin\":\"1.3.1\",\"blockStyle\":\"light\",\"isFirstOnHierarchy\":false,\"uniqueID\":\"column-maxi-1\",\"customLabel\":\"Column_1\",\"column-size-general\":50,\"column-size-m\":100} -->
<div class=\"wp-block-maxi-blocks-column-maxi maxi-block maxi-column-block maxi-light column-maxi-1\" id=\"column-maxi-1\"></div>
<!-- /wp:maxi-blocks/column-maxi -->

<!-- wp:maxi-blocks/column-maxi {\"maxi-version-current\":\"1.3.1\",\"maxi-version-origin\":\"1.3.1\",\"blockStyle\":\"light\",\"isFirstOnHierarchy\":false,\"uniqueID\":\"column-maxi-2\",\"customLabel\":\"Column_2\",\"column-size-general\":50,\"column-size-m\":100} -->
<div class=\"wp-block-maxi-blocks-column-maxi maxi-block maxi-column-block maxi-light column-maxi-2\" id=\"column-maxi-2\"></div>
<!-- /wp:maxi-blocks/column-maxi --></div>
<!-- /wp:maxi-blocks/row-maxi --></section>
<!-- /wp:maxi-blocks/container-maxi -->"
`;

exports[`Row Maxi Row Maxi does not break 2`] = `"<style>body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 1170px; flex-wrap: wrap; row-gap: 20px; column-gap: 2.5%;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xl\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 1170px; flex-wrap: wrap; row-gap: 20px; column-gap: 2.5%;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"l\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"m\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"s\"] .maxi-block.maxi-block--backend.row-maxi-1,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\"xs\"] .maxi-block.maxi-block--backend.row-maxi-1{ max-width: 90%;}</style>"`;
